package.json

```diff
@@ -68,6 +70,21 @@
     "lib": "./lib/mongoose"
   },
   "scripts": {
+    "docs:clean": "npm run docs:clean:stable",
+    "docs:clean:stable": "rimraf index.html && rimraf -rf ./docs/*.html  && rimraf -rf ./docs/api && rimraf -rf ./docs/tutorials/*.html && rimraf -rf ./docs/typescript/*.html && rimraf -rf ./docs/*.html && rimraf -rf ./docs/source/_docs && rimraf -rf ./tmp",
+    "docs:clean:legacy": "rimraf index.html && rimraf -rf ./docs/5.x && rimraf -rf ./docs/source/_docs && rimraf -rf ./tmp",
+    "docs:copy:tmp": "mkdirp ./tmp/docs/css && mkdirp ./tmp/docs/js && mkdirp ./tmp/docs/images && mkdirp ./tmp/docs/tutorials && mkdirp ./tmp/docs/typescript && ncp ./docs/css ./tmp/docs/css --filter=.css$ && ncp ./docs/js ./tmp/docs/js --filter=.js$ && ncp ./docs/images ./tmp/docs/images && ncp ./docs/tutorials ./tmp/docs/tutorials && ncp ./docs/typescript ./tmp/docs/typescript && cp index.html ./tmp",
+    "docs:copy:tmp:legacy": "rimraf ./docs/5.x && ncp ./tmp ./docs/5.x",
+    "docs:checkout:gh-pages": "git checkout gh-pages",
+    "docs:checkout:legacy": "git checkout 5.x",
+    "docs:generate:html": "node website.js",
+    "docs:generate:search": "node docs/search.js",
+    "docs:merge:stable": "git merge master",
+    "docs:merge:legacy": "git merge 5.x",
+    "docs:test": "npm run docs:generate:html && npm run docs:generate:search",
+    "docs:view": "node website.js && node static.js",
+    "docs:prepare:publish:stable": "npm run docs:checkout:gh-pages && docs:merge:stable && npm run docs:clean:stable && npm run docs:generate:html && npm run docs:generate:search",
```

Comment:
Minor nitpick: why not just `docs:publish:stable`? More concise.

Reply:
I thought of prepare, because it does not really publish the docs. publishing would be merging it into gh-pages branch.